sdk/core/azure-core-tracing-opencensus/src/test/java/com/azure/core/tracing/opencensus/OpenCensusTracerTest.java

```diff
@@ -50,27 +200,99 @@ public void startSpanOverloadNullPointerException() {
         openCensusTracer.start("", Context.NONE, null);
     }
 
+    //add tests for number of child spans created and how parent span relation
+
     @Test
-    public void startSpanSendOperation() {
+    public void addLinkTest() {
+        // Arrange
+        // Create a child-parent link between multiple spans
+        final RecordEventsSpanImpl testSpan =
+            (RecordEventsSpanImpl) tracer.spanBuilder("new-test-span").startSpan();
+        final Context traceContext = tracingContext.addData(SPAN_CONTEXT_KEY, testSpan.getContext());
+        final RecordEventsSpanImpl parentSpanImpl = (RecordEventsSpanImpl) parentSpan;
+
         // Act
-        final Context updatedContext = openCensusTracer.start(METHOD_NAME, Context.NONE, ProcessKind.SEND);
+        openCensusTracer.addLink(traceContext);
+
+        //Assert
+        // verify both spans share the same link traceId
+        Assert.assertEquals(parentSpanImpl.toSpanData().getContext().getTraceId(),
```

Comment:
this seems to be an incorrect validation. Links do not flow into trace-id.

So i guess what happens in this test:
setUp creates parent span
testSpan will get same TraceId - because it's a child
traceContext gets context from traceSpan
link is created from traceContext (which shares trace-id with parentSpan and testSpan
Then you check if parentSpan has the same traceId as link

What should be tested:
link should have random context - new one
you add it  to the span
and then compare links on the parent span to what you've added

Reply:
updated.